File: model/src/main/java/net/artux/pda/model/chat/ChatUpdate.java
package net.artux.pda.model.chat;

import java.time.Instant;
import java.util.List;
import java.util.stream.Collectors;

import lombok.Data;

@Data
public class ChatUpdate {

    private List<UserMessage> updates;
    private List<ChatEvent> events;
    private Instant timestamp;

    public List<UserMessage> getUpdatesByType(UserMessage.Type type) {
        return getUpdates()
                .stream()
                .filter(msg -> msg.getType() == type)
                .collect(Collectors.toList());
    }

}

File: model/src/main/java/net/artux/pda/model/chat/ChatUpdate.kt
package net.artux.pda.model.chat

import java.time.Instant
import java.util.stream.Collectors

class ChatUpdate {
    var updates: List<UserMessage> = mutableListOf()
    var events: List<ChatEvent>? = null
    var timestamp: Instant? = null
    fun getUpdatesByType(type: UserMessage.Type): List<UserMessage> {
        return updates
            .stream()
            .filter { msg: UserMessage -> msg.type === type }
            .collect(Collectors.toList())
    }
}